pkg/fqdn/dnsproxy/udp.go

```diff
@@ -126,10 +138,10 @@ func (f *sessionUDPFactory) SetSocketOptions(conn *net.UDPConn) error {
 	//   checking that a route exists from the source address before
 	//   the source address is replaced with the (transparently) changed one
 	udpOnce.Do(func() {
-		rawconn4 = bindUDP("127.0.0.1") // raw socket for sending IPv4
-		rawconn6 = bindUDP("::1")       // raw socket for sending IPv6
+		rawconn4 = bindUDP("127.0.0.1", f.ipv4Enabled, f.ipv6Enabled) // raw socket for sending IPv4
+		rawconn6 = bindUDP("::1", f.ipv4Enabled, f.ipv6Enabled)       // raw socket for sending IPv6
```

Comment:
@jrajahalme The comments for `transparentSetsockopt` suggest that either socket may receive traffic in either address family. Is this correct? (and does it need comments in case someone else ever debugs it?)